### ISSUE TITLE
symfony/console and unicode text on Win10 console

### ISSUE THREAD

**Issue by aleksandr-shevchenko** [symfony/symfony#49942]
### Symfony version(s) affected

6.2.8

### Description

`echo` can print unicode text to windows console, but symfony/console not.

### How to reproduce

I found a place where the windows console gets corrupted and it can no longer display unicode text. Please fix it.

symfony\console\Terminal.php:
```
    private static function initDimensions()
    {
        echo "before initDimensions() - йфяЙФЯёЁ üÜiİöÖğĞıIəƏçÇşŞ" . PHP_EOL;
        if ('\\' === \DIRECTORY_SEPARATOR) {
            if (preg_match('/^(\d+)x(\d+)(?: \((\d+)x(\d+)\))?$/', trim(getenv('ANSICON')), $matches)) {
                // extract [w, H] from "wxh (WxH)"
                // or [w, h] from "wxh"
                self::$width = (int) $matches[1];
                self::$height = isset($matches[4]) ? (int) $matches[4] : (int) $matches[2];
            } elseif (!self::hasVt100Support() && self::hasSttyAvailable()) {
                // only use stty on Windows if the terminal does not support vt100 (e.g. Windows 7 + git-bash)
                // testing for stty in a Windows 10 vt100-enabled console will implicitly disable vt100 support on STDOUT
                self::initDimensionsUsingStty();
            } elseif (null !== $dimensions = self::getConsoleMode()) {
                // extract [w, h] from "wxh"
                self::$width = (int) $dimensions[0];
                self::$height = (int) $dimensions[1];
            }
        } else {
            self::initDimensionsUsingStty();
        }
        echo "after initDimensions() -- йфяЙФЯёЁ üÜiİöÖğĞıIəƏçÇşŞ" . PHP_EOL;
    }
```
<img width="858" alt="ss_2023_04_05__15_28_10" src="https://user-images.githubusercontent.com/25248634/230067636-4a3b38aa-10e3-49f5-b59e-df2dd74850aa.png">

**Comment by stof** [comment]
which OS are you using (to know which branch of this code is actually executed) ?

**Comment by aleksandr-shevchenko** [comment]
Windows 10

**Comment by maxbeckers** [comment]
I cannot reproduce this bug,
On git-bash using windows 10:
![Screenshot 2023-04-17 120129](https://user-images.githubusercontent.com/11738128/232454199-6d66732b-f745-47ed-bd37-c1a4f01b7cfb.png)

And on CMD:
![Screenshot 2023-04-17 120213](https://user-images.githubusercontent.com/11738128/232454262-0a5730d9-1336-4acd-b771-192fb17890f4.png)

**Comment by aleksandr-shevchenko** [comment]
@maxbeckers
The broken encoding bug has been fixed in 6.2.x-dev (see https://github.com/symfony/symfony/pull/49987).
Try <=6.2.8

**Comment by maxbeckers** [comment]
Ah, thanks ... then this issue can be closed IMHO. //cc @nicolas-grekas
